designs/air-gapped.md

```diff
@@ -0,0 +1,97 @@
+
+## Introduction
+
+**Problem:** we offer a set features that allow users to run EKS-A cluster in disconnected environments, but they lack cohesion.
+
+Overtime, we have developed a particular set of features and tools that, even though they were aimed to solve independent problems, they all converge to offer to support for running EKS-A in environments without Internet connection.
+As a result of their independent development, their interfaces and behaviors are not in sync, which makes the user experience not ideal.
+
+This document aims to solve that, learning from the experience gained building such tools and offering a more cohesive and simple to use solution, without drastically changing the backbone of those features and how they interact with each other.
+
+### Tenets
+
+* ****Simple:**** minimal interface, minimum number of manual steps.
+
+### Goals and Objectives
+
+As a EKS-A cluster administrator I want to:
+
+* Manage clusters in an environment without Internet connection.
+* Populate my environment with the necessary dependencies without having to worry about what those dependencies are and with minimal effort.
+* Interact with EKS-A clusters in air-gapped environments in the same way I interact with any other EKS-A cluster.
+
+### Statement of Scope
+
+**In scope**
+* Download dependencies and package them.
+* Populate disconnected environments with dependencies without Internet connection.
+
+**Not in scope**
+* Setup additional infrastructure to support disconnected environments (eg. private registries).
+
+**Future scope**
+* Package dependencies selectively.
+* Populate disconnected environments with dependencies with Internet connection.
+ 
+## Current state
+In order to support disconnected environments, users currently have to:
+1. Provision an OCI registry.
+1. Run `eksctl anywhere download artifacts` to download manifests to disk.
+1. Run `eksctl anywhere download images` to download images and charts to disk.
+1. Run `eksctl anywhere import images` to push container images and charts to a registry.
+1. Configure their cluster spec file to point to the registry 
+1. Run all cluster commands (`create`, `upgrade`, etc.) with the `--bundles-override` flag pointing to the `Bundles` manifest downloaded in the first step.
+
+## Overview of Solution
+1. Users will provide an OCI registry.
+1. We will store all dependencies (container images, helm charts and yaml manifests) in that registry.
+1. If a cluster config with a [`registryMirrorConfiguration`](https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cluster_types.go#L51) is provided, the CLI and the rest of components will always pull them from it.
+No extra flags/configuration will be needed and the user will interact with this cluster as with any other one.
+
+## Solution Details
+
+### Populating the registry
+
+In order to populate the registry, we will offer two commands:
+```sh
+eksctl anywhere export artifacts --output artifacts.tar
```

Comment:
how about `download artifacts` and `upload artifacts`?

Reply:
I look at it from the other end, I think export/import is more generic.
Download/upload implies there is a network connection. Which for now, it is true for all 3 artifacts. But this might not be true for other types of artifacts we might add in the future (imagine something that we just store in disk in a particular location).

The idea behind this proposal is to be able to be flexible enough to accommodate future expansions. Hence, my preference for a more generic export/import.

Reply:
If we have the ability to directly export to a registry in the future in one command, what would you suggest calling that command? Or is the plan to just have users run both?

Reply:
That would depend on use feedback.

Two commands are necessary to support "truly disconnected" environments where there not one machine that can act as a bridge between the internet and the registry.

I'm not sure if it's worth the effort since they can just chain the commands with `&&`, but if there are enough folks who want one single command, we could build it. As far as the name, I think we can wait until we decide to build it. But maybe: `populate artifacts`, `export-import artifacts`...